File: 0366_Find Leaves of Binary Tree/FindLeavesofBinaryTree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public IList<IList<int>> FindLeaves(TreeNode root) {
        var ans = new List<IList<int>>();
        Height(root, ans);
        return ans;
    }
    
    private int Height(TreeNode node, IList<IList<int>> ans)
    {
        if(node == null) return -1;
        var height = 0;
        height = Math.Max(Height(node.left, ans), Height(node.right, ans)) + 1;
        if(ans.Count == height)
            ans.Add(new List<int>());
        ans[height].Add(node.val);
        return height;
    }
}

File: 0366_Find Leaves of Binary Tree/FindLeavesofBinaryTree.java
class Solution {
    public List<List<Integer>> findLeaves(TreeNode root) {
        List<List<Integer>> ans = new ArrayList<List<Integer>>();
        height(root, ans);
        return ans;
    }
    
    private int height(TreeNode node, List<List<Integer>> ans) {
        if(node == null) return -1;
        int height = Math.max(height(node.left, ans), height(node.right, ans)) + 1;
        if(ans.size() == height) ans.add(new ArrayList<Integer>());
        ans.get(height).add(node.val);
        return height;
    }
}